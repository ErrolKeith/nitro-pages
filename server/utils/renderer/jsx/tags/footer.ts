import { makeJsx } from "./makeJsx";

export interface FooterConfig {
  props: { className?: string };
  children: React.ReactElement[];
}

export function makeFooter(config: FooterConfig) {
  return makeJsx("footer", config.props, config.children);
}
