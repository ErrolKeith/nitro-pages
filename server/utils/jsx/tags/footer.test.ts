import { it, expect } from "vitest";
import { makeFooter, FooterConfig } from "./footer";

it("can make a footer tag", () => {
  const footerTestConfig: FooterConfig = {
    props: {
      className: "footer-class",
    },
    children: [],
  };
  const footerTest = makeFooter(footerTestConfig);
  expect(footerTest).toMatchSnapshot();
});
