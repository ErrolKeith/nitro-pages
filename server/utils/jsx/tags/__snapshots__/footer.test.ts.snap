// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can make a footer tag 1`] = `
<footer
  className="footer-class"
/>
`;
